imgui-SFML.cpp

```diff
@@ -141,8 +128,34 @@ struct StickInfo {
     float threshold;
 };
 
-StickInfo s_dPadInfo;
-StickInfo s_lStickInfo;
+struct WindowContext {
+    sf::Window* window = nullptr;
+    ImGuiContext* im_ctx = nullptr;
+
+    bool windowHasFocus = false;
+    bool mousePressed[3] = {false, false, false};
+    bool touchDown[3] = {false, false, false};
+    bool mouseMoved = false;
+    sf::Vector2i touchPos;
+    sf::Texture* fontTexture = nullptr; // owning pointer to internal font atlas which is used if user
+                                        // doesn't set custom sf::Texture.
+    unsigned int joystickId = NULL_JOYSTICK_ID;
+    unsigned int joystickMapping[ImGuiNavInput_COUNT];
+    StickInfo dPadInfo;
+    StickInfo lStickInfo;
+
+    sf::Cursor*mouseCursors[ImGuiMouseCursor_COUNT];
+    bool mouseCursorLoaded[ImGuiMouseCursor_COUNT];
+
+#ifdef ANDROID
+#ifdef USE_JNI
+    bool wantTextInput = false;
+#endif
+#endif
+};
+
+std::vector<WindowContext> windowContexts;
+WindowContext* s_currentWindowContext = nullptr;
```

Comment:
I think this should be called "s_currWindowCtx" so that the code which uses it takes less space.